azafea/event_processors/endless/ping/v1/cli.py

```diff
@@ -122,3 +132,27 @@ def do_parse_images(config: Config, args: argparse.Namespace) -> None:
     progress(num_records, num_records, end='\n')
 
     log.info('All done!')
+
+
+def do_transform_countries_alpha_3_to_2(config: Config, args: argparse.Namespace) -> None:
+    db = Db(config.postgresql)
+    log.info('Tranform alpha-3 country codes into alpha-2 for pings')
+
+    with db as dbsession:
+        query = dbsession.chunked_query(Ping, chunk_size=args.chunk_size)
+        query = query.filter(func.length(Ping.country) == 3)
+        num_records = query.count()
+
+        if num_records == 0:
+            log.info('-> No ping with alpha-3 country code found')
+            return None
+
+        for chunk_number, chunk in enumerate(query, start=1):
+            for ping in chunk:
+                ping.country = tranform_alpha_3_into_2(ping.country)
+            dbsession.commit()
+            progress(chunk_number * args.chunk_size, num_records)
```

Comment:
Didn't we have an issue with another similar migration recently where this proved extremely slow? Might it perform better to do something like (pseudocode):

```
for alpha_3, alpha_2 in COUNTRIES.items():
    dbsession.execute("UPDATE activation_v1 SET country = ? WHERE country = ?", alpha_2, alpha_3)
```

which would allow the work to live in the database, rather than having to stream all 23 million rows (in chunks) in and out of the database?

Reply:
I suppose we can try it and see.

Reply:
Oh, I see @adarnimrod  had the same observation above. I would naively expect the `UPDATE` to perform better. I could be wrong!

Reply:
> I suppose we can try it and see.

I can try on the test database.

> tranform_alpha_3_into_2

Oops, there’s a typo…

Reply:
The migration takes about 3.5 days, it’s too long.

We can generate one request for each country value, that’s dirty but should be fast enough. I can try to find a better solution if you prefer.

Reply:
I think one UPDATE per country code is a clean solution, personally :)

Reply:
> I think one UPDATE per country code is a clean solution, personally :)

:smile:

The new solution takes 10 minutes for pings. Tests pass, the result seems to be OK, but I’d prefer to have another review, just to be sure…